### PR TITLE
Disable slot editing

### PR DESCRIPTION
**Pull request by dwhenry**
These changes are designed to make the slot selection process easier to use/understand based on the lab.

Changes to slot selection:

* Move time selection to above current slot being selected
* Disabled editing slot - this now requires a slot to be removed and the correct slot added
* Add alert message once all slots have been filled and user is attempting to select another day

This functionality can be reviewed and tested here: https://ggp-pensionwise-booking-slots.herokuapp.com/locations/183080c6-642b-4b8f-96fd-891f5cd9f9c7/booking-request/step-one

### CODE REVIEW

File: dist/javascripts/moj.slot-picker.js [hunk 1]
```diff
@@ -26,7 +26,7 @@
   };
 
   SlotPicker.prototype = {
-
```
Comment (benbarnett) [inline dist/javascripts/moj.slot-picker.js:29]: Is this intentional?

File: dist/javascripts/moj.slot-picker.js [hunk 2]
```diff
@@ -86,6 +91,11 @@
 
       this.$_el.on('click chosen', '.BookingCalendar-dateLink, .DateSlider-largeDates li', function(e) {
         e.preventDefault();
+        if(self.limitReached()) {
```
Comment (benbarnett) [inline dist/javascripts/moj.slot-picker.js]: I don't know if we have linting set up but this whole block could do with spaces around the brackets and dropping blocks onto their own lines

**Review by benbarnett — COMMENTED** [review]
As much as the whitespace cleanups are nice, it makes the diff a little fiddly. Could we address those in a separate PR?

**Comment by benbarnett** [comment]
I think this is working really well functionally now!

File: dist/javascripts/moj.slot-picker.js (hunk 2 above)
Comment (dwhenry) [reply dist/javascripts/moj.slot-picker.js]: It does have linting so not sure why this wasn't caught.

**Comment by dwhenry** [comment]
@benbarnett should be ready to review again - will add whitespace changes to a separate PR.

File: dist/javascripts/moj.slot-picker.js [hunk 3]
```diff
@@ -106,11 +118,24 @@
...
         self.confirmVisibility($('.DateSlider').first(), 'top');
       });
+    },
 
-      this.$_el.on('click', '.SlotPicker-choice.is-chosen', function() {
-        var date = $(this).find('.SlotPicker-icon--remove').data('slot-option').attr('id').split('slot-')[1].substr(0, 10);
-        $('.BookingCalendar-dateLink[data-date="' + date + '"]').click();
-      });
+    moveTimeSlotsToChoice: function() {
+      var self = this;
```
Comment (benbarnett) [inline dist/javascripts/moj.slot-picker.js]: Not sure this line is needed in this method

**Review by benbarnett — APPROVED** [review]
Looks good other than that one small comment!

**Pull request merged**